### PR TITLE
BHBC-1841: Remove Project/Survey Publish Buttons And Related Code

### PR DESCRIPTION
**Pull request by curtisupshall**
# Overview

Project and survey metadata no longer needs to be 'published' It should be available 'publicly' immediately upon creation

## Links to Jira tickets

https://quartech.atlassian.net/browse/BHBC-1841

## Description of relevant changes

 - Project/survey UI no longer includes publish button
 - API no longer has endpoints for publishing

### CODE REVIEW

File: database/src/migrations/release.0.34/biohub.sql [hunk 1]
```diff
@@ -1014,7 +1014,7 @@ COMMENT ON COLUMN project.coordinator_agency_name IS 'Name of agency the project
 ;
 COMMENT ON COLUMN project.coordinator_public IS 'A flag that determines whether personal coordinator details are public. A value of "Y" provides that personal details are public.'
 ;
-COMMENT ON COLUMN project.publish_timestamp IS 'A timestamp that indicates that the project metadata has been approved for discovery. If the timestamp is not null then project metadata is public. If the timestamp is null the project metadata is not yet public.'
+COMMENT ON COLUMN project.publish_timestamp IS '(Depricated) A timestamp that indicates that the project metadata has been approved for discovery. If the timestamp is not null then project metadata is public. If the timestamp is null the project metadata is not yet public.'
```
Comment (NickPhura) [inline database/src/migrations/release.0.34/biohub.sql]: nit spelling: `Deprecated`

File: api/src/services/eml-service.ts [hunk 2]
```diff
@@ -251,7 +251,6 @@ export class EmlService extends DBService {
         $: { system: this.constants.EML_PROVIDER_URL, id: this.packageId },
         title: options?.datasetTitle || this.projectData.project.project_name,
         creator: this.getDatasetCreator(),
-        ...(this.projectData.project.publish_date && { pubDate: this.projectData.project.publish_date }),
```
Comment (NickPhura) [inline api/src/services/eml-service.ts:254]: I think we might have to replace this with another date. As I think the assumption is that the EML is 'published' in that it is automatically publicly consumable.

And we can just semi hard code it.

```
...
creator: this.getDatasetCreator(),
pubDate: moment().toISOString(),
metadataProvider: {
...
```

File: app/src/features/surveys/view/SurveyHeader.tsx [hunk 3]
```diff
@@ -279,14 +240,6 @@ const SurveyHeader: React.FC<ISurveyHeaderProps> = (props) => {
...
             </Box>
             <Box ml={4} mb={4}>
-              <Button
-                variant="outlined"
-                color="primary"
-                className={classes.actionButton}
-                data-testid="publish-survey-button"
-                onClick={() => togglePublishSurvey()}>
-                {surveyWithDetails.surveyData.survey_details.publish_date ? 'Unpublish Survey' : 'Publish Survey'}
-              </Button>
```
Comment (NickPhura) [inline app/src/features/surveys/view/SurveyHeader.tsx:289]: (not this line, wont let me comment on the right one) 

On line 294 below, the delete tooltip still references not being able to delete a published survey.

Comment (NickPhura) [reply app/src/features/surveys/view/SurveyHeader.tsx:289]: Can probably just drop the tooltip wrapper altogether (we dont have one for the project delete button)

**Comment by NickPhura** [comment]
There are some comments that still reference the concept of published data that we should clean up. I know where most of them are so I could commit them, but ill wait until we sync up in discord later, so I don't cause needless conflicts with any changes you are making.

File: api/src/services/eml-service.ts (hunk 2 above)
Comment (curtisupshall) [reply api/src/services/eml-service.ts:254]: Thoughts on `new Date().toISOString` vs `moment().toISOString()`?

**Review by NickPhura — DISMISSED** [review]
Nice work!

**Review by NickPhura — APPROVED** [review]

**Pull request merged**